proposals/evaluation/uc-0-effective-acr.md

```diff
@@ -0,0 +1,56 @@
+# Effective Access Control Resource Discovery Use Cases
+
+This is part 0 of the [implementation specific use cases comparison](./use-cases.md).
+
+The resource governing access over another resource is called its effective access control resource (ACR).
+
+## 1. Discovering a resource's effective access control
+
+A Solid client needs to discover effective ACR in order to understand and or edit access permissions.
+
+Both WAC and ACP use the same effective ACR discovery mechanism.
+
+### Setup
+
+We have the following hierarchy of resources:
+
+```
+</x>
+</x.acr>
+```
+
+### Universal effective ACR discovery
+
+An agent making a GET or HEAD HTTP request on `</x>` will receive a `Link: </x.acr>; rel="acl"` header in the response that points to the above `</x.acr>`.
+
+The Link header with relationship type `acl` indicates the access control resource of a resource.
+
+### ACP
+
+In ACP, every resource has exactly 1 effective access control resource directly associated with it, and every access control resource directly governs access over itself and exactly one other resource.
+
+In ACP, access control statements can be spread over several resources; that is, an access control resource can reference other resources.
+
+In ACP, the access control system in place (that is, ACP, as opposed to WAC, for example) is indicated via a Link header of `rel="type"` `<http://www.w3.org/ns/solid/acp#AccessControlResource>` in HTTP response to a request for an ACR.
+
+### WAC
+
+In WAC, the effective ACR of a resource might be the ACR of a parent container of the resource.
+
+In WAC, ACRs are called ACLs (Access Control Lists).
+
+WAC's [Effective ACL Resource](https://solid.github.io/web-access-control-spec/#effective-acl-resource) discovery is described in [the WAC spec](https://solid.github.io/web-access-control-spec/) as follows:
+
+> ### Effective ACL Resource Algorithm
+> To determine the effective ACL resource of a resource, perform the following steps. Returns string (the URI of an ACL Resource).
+>
+> 1. Let resource be the resource.
+> 2. Let aclResource be the ACL resource of resource.
+> 3. If resource has an associated aclResource with a representation, return aclResource.
+> 4. Otherwise, repeat the steps using the container resource of resource.
+
```

Comment:
I don't think the algorithm for determining the ACL of a resource in the spec is relevant to the client. 
In the previous wiki version of the spec on which the current spec is based it was stated quite clearly that one should not try to guess. 
So I think that text is only aimed at the server, to allow the server to determine which ACL to add to the header. Am I right @csarven ?
btw. I think that means the text "To determine the effective ACL..." should be: "For a server to determine..." the effective ACL of a resource...

Reply:
No, it is aimed at any anything that wants to determine the ACL of a resource, including servers, clients..

Reply:
The [previous version of the document](https://github.com/solid/web-access-control-spec/blob/main/README-v0.5.0.md) that was the official one just a little over a month ago, stated that 

> It is considered an anti-pattern for a client to perform those steps, however. A client may discover and load a document's individual resource (for example, for the purposes of editing its permissions). If such a resource does not exist, a client SHOULD NOT search for, or interact with, the inherited ACLs from an upstream container.

Also the algorithm given for finding the container is very vague, because you are trying to specify WAC without assuming LDP, which would make finding the container a lot simpler.

So should we instead write up an example where the client has to try to guess its way back up the hierarchy to find the default ACL? I guess we can make our lives a bit simpler to ourselves if we also have a 
`Link: <.>; rev="http://www.w3.org/ns/ldp#contains">` on the resource, so that the client can find the default.

Which do you prefer?

Reply:
The previous document did say that but we also had an issue on https://github.com/solid/specification/issues/106 to change that. WAC ED's #effective-acl-resource addressed that issue.

As for WAC without LDP's containment.. that makes the spec more flexible and combined with other specs. Servers implementing LDP/Solid can already work with it.